pkg/controller/sync/sync_controller.go

```diff
@@ -116,64 +113,10 @@ func (r *ReconcileSync) Reconcile(request reconcile.Request) (reconcile.Result,
 		return reconcile.Result{}, nil
 	}
 
-	if instance.GetDeletionTimestamp().IsZero() {
-		if !containsString(finalizerName, instance.GetFinalizers()) {
-			instance.SetFinalizers(append(instance.GetFinalizers(), finalizerName))
-			// For some reason the instance sometimes gets changed by update when there is a race
-			// condition that leads to a validating webhook deny of the update
-			cpy := instance.DeepCopy()
-			if err := r.Update(context.Background(), cpy); err != nil {
-				return reconcile.Result{}, err
-			}
-			if !reflect.DeepEqual(instance, cpy) {
-				log.Info("instance and cpy differ")
-			}
-		}
-		log.Info("data will be added", "data", instance)
-		if _, err := r.opa.AddData(context.Background(), instance); err != nil {
-			return reconcile.Result{}, err
-		}
-	} else {
-		// Handle deletion
-		if HasFinalizer(instance) {
```

Comment:
We should still handle the case where deletion timestamp is set by calling `opa.RemoveData()`

This covers the case where the resource is around but has not yet been GC'd (ex. b/c there is some other finalizer set)